crates/router/src/core/payments.rs

```diff
@@ -387,17 +387,23 @@ where
                 mandate_type,
             )
             .await?;
-        operation
-            .to_domain()?
-            .call_unified_authentication_service_if_eligible(
-                state,
-                &mut payment_data,
-                &mut should_continue_transaction,
-                &connector_details,
-                &business_profile,
-                &key_store,
-            )
-            .await?;
+
+        if let Some(ref authentication_product_ids) = business_profile.authentication_product_ids {
```

Comment:
This check should happen in `call_unified_authentication_service_if_eligible`

Reply:
+1